saleor/attribute/models/page.py

```diff
@@ -24,6 +33,10 @@ class Meta:
     def get_ordering_queryset(self):
         return self.assignment.pagevalueassignment.all()
 
+    def save(self, *args, **kwargs):
+        self.page = self.assignment.page
+        super(AssignedPageAttributeValue, self).save(*args, **kwargs)
```

Comment:
This code adds 1 additional db query per each save.

Reply:
fixed by changing the strategy, right now there is one place where page from AssignedPageAttributeValue is updated, no need for signals and override save method.